### ISSUE TITLE
Doesn't work with Fritz.box Fon WLAN 7170

### ISSUE THREAD

**Issue by jogibear9988** [iobroker-community-adapters/ioBroker.tr-064-community#19]
_From @Jey-Cee on March 5, 2018 13:13_

The Fritz.box Fon WLAN 7170 has tr-064 support, but the Adapter can not connect to it.

`tr-064.1 | 2018-03-05 14:08:11.805 | error | ~
-- | -- | -- | --
tr-064.1 | 2018-03-05 14:08:11.805 | error | ~~ If configuration, networt, IP address, etc. ok, try to restart your FritzBox
tr-064.1 | 2018-03-05 14:08:11.804 | error | ~~ Fatal error. Can not connect to your FritzBox.
tr-064.1 | 2018-03-05 14:08:11.804 | error | ~
tr-064.1 | 2018-03-05 14:08:11.804 | error | Error: sendSOAPActionRequest Error action=GetInfo serviceType=urn:dslforum-org:service:WLANConfiguration:1: 401 - {"code":401}
`

Here is the tr64desc.xml:

  | <?xml version="1.0"?>
-- | --
  | <root xmlns="urn:dslforum-org:device-1-0">
  | <specVersion>
  | <major>1</major>
  | <minor>0</minor>
  | </specVersion>
  | <device>
  | <deviceType>urn:dslforum-org:device:InternetGatewayDevice:1</deviceType>
  | <friendlyName>FRITZ!Box Fon WLAN 7170 UPnP/1.0 AVM FRITZ!Box Fon WLAN 7170 (UI) 29.04.88</friendlyName>
  | <manufacturer>AVM</manufacturer>
  | <manufacturerURL>www.avm.de</manufacturerURL>
  | <modelDescription>FRITZ!Box Fon WLAN 7170 (UI)</modelDescription>
  | <modelName>FRITZ!Box Fon WLAN 7170 (UI)</modelName>
  | <modelNumber> - 1und1</modelNumber>
  | <modelURL>www.avm.de</modelURL>
  | <UDN>uuid:739f2409-bccb-40e7-8e6c-00150C888D1C</UDN>
  | <iconList>
  | <icon>
  | <mimetype>image/gif</mimetype>
  | <width>118</width>
  | <height>119</height>
  | <depth>8</depth>
  | <url>/ligd.gif</url>
  | </icon>
  | </iconList>
  | <serviceList>
  | <service>
  | <serviceType>urn:dslforum-org:service:DeviceInfo:1</serviceType>
  | <serviceId>urn:DeviceInfo-com:serviceId:DeviceInfo1</serviceId>
  | <controlURL>/upnp/control/deviceinfo</controlURL>
  | <eventSubURL>/upnp/control/deviceinfo</eventSubURL>
  | <SCPDURL>/deviceinfoSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:DeviceConfig:1</serviceType>
  | <serviceId>urn:DeviceConfig-com:serviceId:DeviceConfig1</serviceId>
  | <controlURL>/upnp/control/deviceconfig</controlURL>
  | <eventSubURL>/upnp/control/deviceconfig</eventSubURL>
  | <SCPDURL>/deviceconfigSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:Layer3Forwarding:1</serviceType>
  | <serviceId>urn:Layer3Forwarding-com:serviceId:Layer3Forwarding1</serviceId>
  | <controlURL>/upnp/control/layer3forwarding</controlURL>
  | <eventSubURL>/upnp/control/layer3forwarding</eventSubURL>
  | <SCPDURL>/layer3forwardingSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:LANConfigSecurity:1</serviceType>
  | <serviceId>urn:LANConfigSecurity-com:serviceId:LANConfigSecurity1</serviceId>
  | <controlURL>/upnp/control/lanconfigsecurity</controlURL>
  | <eventSubURL>/upnp/control/lanconfigsecurity</eventSubURL>
  | <SCPDURL>/lanconfigsecuritySCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:ManagementServer:1</serviceType>
  | <serviceId>urn:ManagementServer-com:serviceId:ManagementServer1</serviceId>
  | <controlURL>/upnp/control/mgmsrv</controlURL>
  | <eventSubURL>/upnp/control/mgmsrv</eventSubURL>
  | <SCPDURL>/mgmsrvSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:Time:1</serviceType>
  | <serviceId>urn:Time-com:serviceId:Time1</serviceId>
  | <controlURL>/upnp/control/time</controlURL>
  | <eventSubURL>/upnp/control/time</eventSubURL>
  | <SCPDURL>/timeSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:UserInterface:1</serviceType>
  | <serviceId>urn:UserInterface-com:serviceId:UserInterface1</serviceId>
  | <controlURL>/upnp/control/userif</controlURL>
  | <eventSubURL>/upnp/control/userif</eventSubURL>
  | <SCPDURL>/userifSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:X_VoIP:1</serviceType>
  | <serviceId>urn:X_VoIP-com:serviceId:X_VoIP1</serviceId>
  | <controlURL>/upnp/control/x_voip</controlURL>
  | <eventSubURL>/upnp/control/x_voip</eventSubURL>
  | <SCPDURL>/x_voipSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:X_AVM-DE_Storage:1</serviceType>
  | <serviceId>urn:X_AVM-DE_Storage-com:serviceId:X_AVM-DE_Storage1</serviceId>
  | <controlURL>/upnp/control/x_storage</controlURL>
  | <eventSubURL>/upnp/control/x_storage</eventSubURL>
  | <SCPDURL>/x_storageSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:X_AVM-DE_OnTel:1</serviceType>
  | <serviceId>urn:X_AVM-DE_OnTel-com:serviceId:X_AVM-DE_OnTel1</serviceId>
  | <controlURL>/upnp/control/x_contact</controlURL>
  | <eventSubURL>/upnp/control/x_contact</eventSubURL>
  | <SCPDURL>/x_contactSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:X_AVM-DE_WebDAVClient:1</serviceType>
  | <serviceId>urn:X_AVM-DE_WebDAV-com:serviceId:X_AVM-DE_WebDAVClient1</serviceId>
  | <controlURL>/upnp/control/x_webdav</controlURL>
  | <eventSubURL>/upnp/control/x_webdav</eventSubURL>
  | <SCPDURL>/x_webdavSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:X_AVM-DE_UPnP:1</serviceType>
  | <serviceId>urn:X_AVM-DE_UPnP-com:serviceId:X_AVM-DE_UPnP1</serviceId>
  | <controlURL>/upnp/control/x_upnp</controlURL>
  | <eventSubURL>/upnp/control/x_upnp</eventSubURL>
  | <SCPDURL>/x_upnpSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:X_AVM-DE_TAM:1</serviceType>
  | <serviceId>urn:X_AVM-DE_TAM-com:serviceId:X_AVM-DE_TAM1</serviceId>
  | <controlURL>/upnp/control/x_tam</controlURL>
  | <eventSubURL>/upnp/control/x_tam</eventSubURL>
  | <SCPDURL>/x_tamSCPD.xml</SCPDURL>
  | </service>
  | </serviceList>
  | <deviceList>
  | <device>
  | <deviceType>urn:dslforum-org:device:LANDevice:1</deviceType>
  | <friendlyName>LANDevice - FRITZ!Box Fon WLAN 7170 (UI)</friendlyName>
  | <manufacturer>AVM</manufacturer>
  | <manufacturerURL>www.avm.de</manufacturerURL>
  | <modelDescription>LANDevice - FRITZ!Box Fon WLAN 7170 (UI)</modelDescription>
  | <modelName>LANDevice - FRITZ!Box Fon WLAN 7170 (UI)</modelName>
  | <modelNumber> - 1und1</modelNumber>
  | <modelURL>www.avm.de</modelURL>
  | <UDN>uuid:75802409-bccb-40e7-8e6b-00150C888D1C</UDN>
  | <UPC>AVM TR-064</UPC>
  | <serviceList>
  | <service>
  | <serviceType>urn:dslforum-org:service:WLANConfiguration:1</serviceType>
  | <serviceId>urn:WLANConfiguration-com:serviceId:WLANConfiguration1</serviceId>
  | <controlURL>/upnp/control/wlanconfig1</controlURL>
  | <eventSubURL>/upnp/control/wlanconfig1</eventSubURL>
  | <SCPDURL>/wlanconfigSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:Hosts:1</serviceType>
  | <serviceId>urn:LanDeviceHosts-com:serviceId:Hosts1</serviceId>
  | <controlURL>/upnp/control/hosts</controlURL>
  | <eventSubURL>/upnp/control/hosts</eventSubURL>
  | <SCPDURL>/hostsSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:LANEthernetInterfaceConfig:1</serviceType>
  | <serviceId>urn:LANEthernetIfCfg-com:serviceId:LANEthernetInterfaceConfig1</serviceId>
  | <controlURL>/upnp/control/lanethernetifcfg</controlURL>
  | <eventSubURL>/upnp/control/lanethernetifcfg</eventSubURL>
  | <SCPDURL>/ethifconfigSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:LANHostConfigManagement:1</serviceType>
  | <serviceId>urn:LANHCfgMgm-com:serviceId:LANHostConfigManagement1</serviceId>
  | <controlURL>/upnp/control/lanhostconfigmgm</controlURL>
  | <eventSubURL>/upnp/control/lanhostconfigmgm</eventSubURL>
  | <SCPDURL>/lanhostconfigmgmSCPD.xml</SCPDURL>
  | </service>
  | </serviceList>
  | </device>
  | <device>
  | <deviceType>urn:dslforum-org:device:WANDevice:1</deviceType>
  | <friendlyName>WANDevice - FRITZ!Box Fon WLAN 7170 (UI)</friendlyName>
  | <manufacturer>AVM</manufacturer>
  | <manufacturerURL>www.avm.de</manufacturerURL>
  | <modelDescription>WANDevice - FRITZ!Box Fon WLAN 7170 (UI)</modelDescription>
  | <modelName>WANDevice - FRITZ!Box Fon WLAN 7170 (UI)</modelName>
  | <modelNumber> - 1und1</modelNumber>
  | <modelURL>www.avm.de</modelURL>
  | <UDN>uuid:75802409-bccb-40e7-8e6a-00150C888D1C</UDN>
  | <UPC>AVM TR-064</UPC>
  | <serviceList>
  | <service>
  | <serviceType>urn:dslforum-org:service:WANCommonInterfaceConfig:1</serviceType>
  | <serviceId>urn:WANCIfConfig-com:serviceId:WANCommonInterfaceConfig1</serviceId>
  | <controlURL>/upnp/control/wancommonifconfig1</controlURL>
  | <eventSubURL>/upnp/control/wancommonifconfig1</eventSubURL>
  | <SCPDURL>/wancommonifconfigSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:WANDSLInterfaceConfig:1</serviceType>
  | <serviceId>urn:WANDSLIfConfig-com:serviceId:WANDSLInterfaceConfig1</serviceId>
  | <controlURL>/upnp/control/wandslifconfig1</controlURL>
  | <eventSubURL>/upnp/control/wandslifconfig1</eventSubURL>
  | <SCPDURL>/wandslifconfigSCPD.xml</SCPDURL>
  | </service>
  | </serviceList>
  | <deviceList>
  | <device>
  | <deviceType>urn:dslforum-org:device:WANConnectionDevice:1</deviceType>
  | <friendlyName>WANConnectionDevice - FRITZ!Box Fon WLAN 7170 (UI)</friendlyName>
  | <manufacturer>AVM</manufacturer>
  | <manufacturerURL>www.avm.de</manufacturerURL>
  | <modelDescription>WANConnectionDevice - FRITZ!Box Fon WLAN 7170 (UI)</modelDescription>
  | <modelName>WANConnectionDevice - FRITZ!Box Fon WLAN 7170 (UI)</modelName>
  | <modelNumber> - 1und1</modelNumber>
  | <modelURL>www.avm.de</modelURL>
  | <UDN>uuid:75802409-bccb-40e7-8e69-00150C888D1C</UDN>
  | <UPC>AVM TR-064</UPC>
  | <serviceList>
  | <service>
  | <serviceType>urn:dslforum-org:service:WANDSLLinkConfig:1</serviceType>
  | <serviceId>urn:WANDSLLinkConfig-com:serviceId:WANDSLLinkConfig1</serviceId>
  | <controlURL>/upnp/control/wandsllinkconfig1</controlURL>
  | <eventSubURL>/upnp/control/wandsllinkconfig1</eventSubURL>
  | <SCPDURL>/wandsllinkconfigSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:WANEthernetLinkConfig:1</serviceType>
  | <serviceId>urn:WANEthernetLinkConfig-com:serviceId:WANEthernetLinkConfig1</serviceId>
  | <controlURL>/upnp/control/wanethlinkconfig1</controlURL>
  | <eventSubURL>/upnp/control/wanethlinkconfig1</eventSubURL>
  | <SCPDURL>/wanethlinkconfigSCPD.xml</SCPDURL>
  | </service>
  | <service>
  | <serviceType>urn:dslforum-org:service:WANPPPConnection:1</serviceType>
  | <serviceId>urn:WANPPPConnection-com:serviceId:WANPPPConnection1</serviceId>
  | <controlURL>/upnp/control/wanpppconn1</controlURL>
  | <eventSubURL>/upnp/control/wanpppconn1</eventSubURL>
  | <SCPDURL>/wanpppconnSCPD.xml</SCPDURL>
  | </service>
  | </serviceList>
  | </device>
  | </deviceList>
  | </device>
  | </deviceList>
  | <presentationURL>http://fritz.box</presentationURL>
  | </device>
  | </root>
  |  

_Copied from original issue: soef/ioBroker.tr-064#68_

**Comment by jogibear9988** [comment]
_From @cimba007 on December 21, 2018 20:52_

Still doesn't work

**Comment by Apollon77** [comment]
Normally 401 means an Auth error ... please check settings and also that the needed access is activated and allowed. Please reopen with more details if all checked

**Comment by padrino-x** [comment]
Just trying an FB7170 and I have to confirm this.
Due to this changelog
ftp://ftp.avm.de/archive/fritz.box/fritzbox.fon_wlan_7170_sl/firmware/deutsch/info.txt
it should support tr-064 since FW 29.04.76.

But what I found so far with "user" and the 7170, is only the topic about "Fernwartung".
I tried to enable that, but no success with the adapter.

I also tried the given IP for the "Fernwartung" and Firefox prompts me with a
"Ein Fehler ist während einer Verbindung mit x.x.x.x aufgetreten. SSL hat einen schwachen kurzlebigen Diffie-Hellman-Schlüssel in der Handshake-Nachricht "Server-Schlüsselaustausch" empfangen. Fehlercode: SSL_ERROR_WEAK_SERVER_EPHEMERAL_DH_KEY"
which it doesn't let me ignore.

Trying with chrome and ignoring its warning, prompts me with the request to login with user and password.
When done I get to the regular "userless" login I normaly see when going to fritz.box.

If this "Fernwartung" is the tr-064 feature, could it be that the too old certificate is a problem for the adapter?
